typespec-extension/src/code-model-builder.ts

```diff
@@ -489,10 +489,6 @@ export class CodeModelBuilder {
     // deduplicate model name
     const nameCount = new Map<string, number>();
     const deduplicateName = (schema: Schema) => {
-      // skip models under "Azure.ResourceManager"
-      if (this.isArm() && schema.language.default?.namespace?.startsWith("Azure.ResourceManager")) {
-        return;
-      }
```

Comment:
This deduplication logic is due to same dummy resource schema being added many times here:
https://github.com/Azure/autorest.java/pull/2749/files#diff-14fdfb2e0df46da414f4d04a218f03e4e5f2737d3a9672b9c4e4feb16419445eL2160